### DISCUSSION TITLE
Allow helmfile state file to be passed via stdin

### DISCUSSION THREAD

**Discussion by ggillies** [helmfile/helmfile#503] [Ideas]
We are looking at using an external tool (jsonnet) to generate a helmfile state file (helmfile.yaml) to pass to helmfile for execution. However, at the moment we have to do something like

```
jsonnet main.jsonnet -o helmfile-from-jsonnet.yaml
helmfile -f helmfile-from-jsonnet.yaml apply
```

It would be really nice if we could have helmfile instead support natively passing it the state file on stdin, so we could instead do

```
jsonnet main.jsonnet | helmfile -f - apply
```

Save us some complexity of creating and cleaning up files on disk.

**Comment by yxxhero** [comment]
@xiaomudk ping

**Comment by xiaomudk** [comment]
Thinks, good idea，Can you create an issue?
